services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/Persistence.kt

```diff
@@ -211,6 +206,22 @@ class HankeEntity(
         }
     }
 
+    fun alkuPvm(): ZonedDateTime? {
+        return listOfHankeAlueet
+            .map { it.haittaAlkuPvm }
+            .filterNotNull()
+            .minOfOrNull { it }
+            ?.atStartOfDay(TZ_UTC)
+    }
+
+    fun loppuPvm(): ZonedDateTime? {
+        return listOfHankeAlueet
+            .map { it.haittaLoppuPvm }
+            .filterNotNull()
```

Comment:
.mapNotNull { it.haittaLoppuPvm }

```suggestion
            .map { it.haittaLoppuPvm }
            .filterNotNull()
```